### PR TITLE
Add get_env jinja function and -e/--env_var option

### LINKED ISSUE #105: Support additional command line arguments
I often find myself specifying flags in the devenv file via environment variables, e.g. I'd have a section like:

```
# development
{% if "PKG_DEV" in os.environ %}
- pytest
- bump2version
{% endif %}
```

for development dependencies that are separate from those which a normal user of the environment needs.

That means that I have to set this environment variable for the dev setup, but I'd much prefer something like this:

    $ conda devenv -a dev

and a function like:

```
{% if "dev" in additional_args() %}
- ...
{% endif %}
```

I'd be open to work on a PR for this.

### PR DESCRIPTION
**Pull request by phausamann**
Implements the features discussed in #105.

Fix #105

### CODE REVIEW

**Comment by phausamann** [comment]
Open questions:
- Should we revert `os.environ` at the end?
- Is there a reason to support env vars whose values contain equal signs?

File: conda_devenv/devenv.py [hunk 1]
```diff
@@ -516,6 +568,13 @@ def main(args=None):
     parser.add_argument(
         "--quiet", action="store_true", default=False, help="Do not show progress"
     )
+    parser.add_argument(
+        "--env_var",
```
Comment (prusse-martin) [inline conda_devenv/devenv.py:572]:
```suggestion
        "--env-var",
```
Is more common for cli options.

**Comment by prusse-martin** [comment]
> * Should we revert `os.environ` at the end?

Maybe not. We can let those envs var "leak" into the `conda env update`  execution IMO.

> * Is there a reason to support env vars whose values contain equal signs?

Good point. Doing `split_arg = arg.split("=", 1)` will handle that.

File: conda_devenv/devenv.py [hunk 2]
```diff
@@ -483,6 +511,30 @@ def get_env_directory(env_name):
...
+    """
+    :param List[str] env_var_args:
+        List of arguments in the form "VAR_NAME" or "VAR_NAME=VALUE"
+    :return: Dict[str,str]
+        Mapping from "VAR_NAME" to "VALUE" or empty str.
+    """
+    env_vars = {}
+    if env_var_args is not None:
+        for arg in env_var_args:
+            split_arg = arg.split("=")
```
Comment (prusse-martin) [inline conda_devenv/devenv.py:524]:
```suggestion
            split_arg = arg.split("=", 1)
```

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -101,6 +101,6 @@ Before you submit a pull request, check that it meets these guidelines:
 2. If the pull request adds functionality, the docs should be updated. Put
    your new functionality into a function with a docstring, and add the
    feature to the list in README.rst.
-3. The pull request should work for Python 2.7, 3.5 and 3.6. Check
-   https://travis-ci.org/ESSS/conda-devenv/pull_requests
+3. The pull request should work for Python 3.6, 3.7 and 3.8. Check
```
Comment (prusse-martin) [inline CONTRIBUTING.rst:104]: BTW.
Thank you for updating this docs.
=)

**Comment by prusse-martin** [comment]
> Actually as mentioned, we just need some docs. 👍

He is probably speaking of the section [Command-line reference in `docs/usage.rst`](https://github.com/ESSS/conda-devenv/blob/master/docs/usage.rst#command-line-reference).

**Comment by nicoddemus** [comment]
> He is probably speaking of the section Command-line reference in docs/usage.rst.

True, but maybe also add a new section below "Checking minimum conda-devenv version" in the README.

**Comment by phausamann** [comment]
Added documentation for `get_env` and the command line option. I've put the new section above the minimum devenv section so that users don't confuse it with the "Environment variables" section.

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -64,7 +64,7 @@ Ready to contribute? Here's how to set up ``conda-devenv`` for local development
 
 3. Create a new conda environment for developing::
 
-    $ conda create -n devenv --file requirements_dev.txt
+    $ conda create -c conda-forge -n devenv --file requirements_dev.txt
```
Comment (prusse-martin) [inline CONTRIBUTING.rst:67]: "conda-forge" channel should not be necessary for this.
Did you have problems with this? Like some missing package?

File: docs/usage.rst [hunk 5]
```diff
@@ -74,6 +74,39 @@ The following variables are available in the Jinja 2 namespace:
      - True if the current file is processed because it was included by another file.
 
 
+Using environment variables in the devenv file
+----------------------------------------------
+
+Environment variables that are defined when calling ``conda devenv`` can be read and checked with the
```
Comment (nicoddemus) [inline docs/usage.rst:80]:
```suggestion
Environment variables that are defined when calling ``conda devenv`` can be read normally using ``os.environ``, but for convenience and better error messages, you can use the
```

**Comment by nicoddemus** [comment]
(We should squash/merge this btw)

**Comment by nicoddemus** [comment]
Thanks a lot @phausamann for the suggestion and very well written PR!

**Comment by nicoddemus** [comment]
Thanks everyone! I will prepare a new release tomorrow. 👍

File: CONTRIBUTING.rst (hunk 4 above)
Comment (phausamann) [reply CONTRIBUTING.rst:67]: Yeah, these are missing when trying to install from defaults on Linux:
```
  - tox
  - bumpversion
  - pre-commit
  - pytest-datadir[version='>=1.0']
```

**Comment by phausamann** [comment]
Happy to help!

**Pull request merged**